lib/phoenix/token.ex

```diff
@@ -0,0 +1,128 @@
+defmodule Phoenix.Token do
+  @moduledoc """
+  Tokens provide a way to  generate, verify, and decrypt bearer
+  tokens for use in Channels or API authentication.
+
+  ## Basic Usage
+  When generating a unique token for usage in an API or Channel
+  it is advised to use a unique identifier for the user typically
+  the id from a Database. For example:
+
+      iex> user_id = 1
+      iex> token = sign_token(endpoint, "user", user_id)
+      iex> user_id == verify_token(endpoint, token)
+      true
+
+  In that example we have a user's id, we generate a token and send
+  it to the client. When the client uses the token in a futher
+  communication you verify the token, query the database and authorize
+  the user.
+
+  When using it with a socket a typical example might be:
+
+    defmodule MyChannel do
+      def join("my:" <> id = topic, %{token: token}, socket) do
+        case verify_token(socket, "user", token) do
+          user_id ->
+            socket = assigns(socket, :user, Repo.get!(User, user_id))
```

Comment:
let's make sure we show a query/check where we authorize the user to access `id`